src/d3a/d3a_core/util.py

```diff
@@ -486,3 +487,28 @@ def is_time_slot_in_past_markets(time_slot: DateTime, current_time_slot: DateTim
             hours=ConstSettings.SettlementMarketSettings.MAX_AGE_SETTLEMENT_MARKET_HOURS))
     else:
         return time_slot < current_time_slot
+
+
+class FutureMarketCounter:
+    """Hold a time counter for the future market.
+
+    In the future market, we only want to clear in a predefined interval.
+    """
+    def __init__(self):
+        self.last_time_dispatched = pendulum.now()
+
+    @property
+    def is_time_for_clearing(self) -> bool:
+        """Compare current time with the latest time clearing was dispatched.
+
+        Returns True if the FUTURE_MARKET_CLEARING_INTERVAL has
+        already passed since the last dispatch time.
+        """
+        current_time = pendulum.now()
```

Comment:
Same as above, you should compare the simulation times and not the actual real times. Therefore it would be better if is_time_for_clearing is a method that receives the current_time as an argument